### ISSUE TITLE
C# example fails to run out of the box (after adding key of course)

### ISSUE THREAD

**Issue by graemeaj** [Azure-Samples/Cognitive-Speech-TTS#168]
I attempted to get the c# example running, but i get a serious error
![image](https://user-images.githubusercontent.com/9042770/75291278-23e8bc00-57df-11ea-9884-a4f9f426d980.png)

"WebException: The request was aborted: Could not create SSL/TLS secure channel."

the endpoint URL (westus.etc) is correct and matches the region in my azure settings.

I see some threads mentioning this error, and it having to do with TLS.

**Comment by boltomli** [comment]
I didn't have the problem here. Do you still hit the problem now?

**Comment by szhaomsft** [comment]
@graemeaj  could u try SDK instead? 

we recommend using SDK for TTS now

https://github.com/Azure-Samples/cognitive-services-speech-sdk

**Comment by graemeaj** [comment]
yeah i never could get it working, so i used the cognitive speech sdk like you linked above

**Comment by johnwood-wff** [comment]
Is the REST API just not supported now? We're getting the same error using the REST API - it seems that the TLS/SSL connection can no longer be established to the endpoints. It makes the service unusable - what's going on?

Just a simple test using the REST API fails, with .Net 4.6.1 using HttpWebRequest POST - get the error: 
A call to SSPI failed, see inner exception.
The message received was unexpected or badly formatted
System.Security.Autthentication.AuthenticationException
   at System.Net.Security.SslState.StartSendAuthResetSignal(ProtocolToken message, AsyncProtocolRequest asyncRequest, Exception exception)

This started a couple of days ago.

**Comment by boltomli** [comment]
If possible can you use HttpClient instead of HttpWebRequest? As I know it's not recommended now in .NET framework.

**Comment by johnwood-wff** [comment]
What in particular does HttpClient do that's different to HttpWebRequest, from the protocol perspective? In this particular case we're doing this through a platform that unfortunately has to retain compatibility with .Net 3.5 where HttpClient isn't available... so we can't really use HttpClient very easily. Are there any other options?

**Comment by boltomli** [comment]
Per my understanding you might need to explicitly specify TLS1.2 for HttpWebRequest, but I don't have actual experience to use it in production. The behaviour may not be the same on different platforms.

**Comment by johnwood-wff** [comment]
Yes I do explicitly set Tls 1.2:

System.Net.ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;

 

I know this is taking effect because if I change it to Tls11 or Tls10 I get a different response that the version of TLS is not supported.

 

And again this was working until a few days ago, so not sure what has happened.

**Comment by boltomli** [comment]
I can run the sample without modification and no errors. I also found that actually it's not necessary to indicate TLS 1.2 explicitly.

But this could depends on the platform you're on. I ran it on Linux through mono, targeting .NET framework 4.5 and 4.6.1 both work. Dotnet core 3.1 works as well, on either Linux and macOS. I believe Windows 10 is OK (Windows 10 was the platform on which the code was developed).
